copy/copy.go

```diff
@@ -1423,6 +1452,12 @@ func (c *copier) copyBlobFromStream(ctx context.Context, srcStream io.Reader, sr
 		default:
 			return types.BlobInfo{}, errors.Errorf("Internal error: Unexpected compressionOperation value %#v", compressionOperation)
 		}
+		if uploadedInfo.Digest != "" && uploadCompressorName != "" && uploadCompressorName != internalblobinfocache.UnknownCompression {
```

Comment:
`uploadedInfo.Digest` can’t be `""` (or we won’t be able to create a manifest).

Reply:
Removing that check.